Fix: prestashop don't display Title of product, category meta title, meta description

Codencode opened the pull request:
Added a check to allow administrators to view the correct meta tags even when the shop is in maintenance mode. Now, if the PS_MAINTENANCE_ALLOW_ADMINS option is enabled and the user is a logged-in administrator, the getMetaTags function will return the appropriate metadata for the requested pages.

<!-----------------------------------------------------------------------------
Thank you for contributing to the PrestaShop project!

Please take the time to edit the "Answers" rows below with the necessary information.

Check out our contribution guidelines to find out how to complete it:
https://devdocs.prestashop-project.org/9/contribute/contribution-guidelines/#pull-requests

For type and category see:
https://devdocs.prestashop-project.org/9/contribute/contribution-guidelines/pull-requests/#type--category
------------------------------------------------------------------------------>

| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | 8.2.x
| Description?      | When new shop is under maintanace mode admin can see FO, but all product, category won't show Title, Meta title, meta description. When you put yourself on list of IP for maintanace it all came back and display correctly. This is nasty thig that I spent 3 hours to fount out on forum that somebody already pos it. But i didn't find this report on Github..
| Type?             | bug fix
| Category?         | FO
| BC breaks?        | no
| Deprecations?     | no
| How to test?      | see https://github.com/PrestaShop/PrestaShop/issues/37954
| UI Tests          | https://github.com/paulnoelcholot/testing_pr/actions/runs/13410555293
| Fixed issue or discussion?     | Fixes #37954
| Related PRs       | 
| Sponsor company   | @Codencode

Hlavtox commented:
@Codencode Just a suggestion, not needed.

What if you created this Tools method:

```
public static function isAllowedToBypassMaintenance()
{
  $is_admin = (int) (new Cookie('psAdmin'))->id_employee;
  $maintenance_allow_admins = (bool) Configuration::get('PS_MAINTENANCE_ALLOW_ADMINS');
  if ($is_admin && $maintenance_allow_admins) {
    return true;
  }

  $allowed_ips = array_map('trim', explode(',', Configuration::get('PS_MAINTENANCE_IP')));
  if (IpUtils::checkIp(Tools::getRemoteAddr(), $allowed_ips)) {
    return true;
  }

  return false;
}
```

You could then use it in your meta code and here https://github.com/PrestaShop/PrestaShop/blob/35b56d8bafe5942bbb524516353268339e949481/classes/controller/FrontController.php#L761
and it won't be duplicated. :-)

Codencode commented:
@Hlavtox,
good idea, so would you create the new method isAllowedToBypassMaintenance in the Tools class?

Hlavtox commented:
@Codencode Yeah yeah, if you don't mind, it would be cleaner. And we would have the code only on one place. :-)

Codencode commented:
@Hlavtox No problem, it's a good solution!
I hadn't thought of that.
Thanks for the suggestion!
I'll make the changes.

Codencode committed:
Improved meta tag handling in maintenance mode

Added a check to allow administrators to view the correct meta tags even when the shop is in maintenance mode. Now, if the PS_MAINTENANCE_ALLOW_ADMINS option is enabled and the user is a logged-in administrator, the getMetaTags function will return the appropriate metadata for the requested pages.

Codencode commented:
@Hlavtox 
I've made the changes and removed the previous commit.
Do you think it's okay?

Hlavtox commented:
@Codencode Yeah just fine, does it work? :-D It was a sketch, didn't test it. :-) Looks good!

Codencode commented:
@Hlavtox 
Yes, of course—I tested it with the following scenarios:
- Maintenance mode active with an IP set
- Maintenance mode active with logged-in access and "Enable store for logged-in employees" enabled
- Maintenance mode active without an IP and with "Enable store for logged-in employees" disabled

The only thing is that this line is repeated both in the new method and in `FrontController::displayMaintenancePage()`, but it's unavoidable.

kpodemski reviewed (approved):
nice 👌🏻

SiraDIOP commented:
Hello @Codencode ,

Thanks for the PR !

I've tested it and it's good for me.
<img width="1440" alt="Capture d’écran 2025-02-20 à 17 28 44" src="https://github.com/user-attachments/assets/50e6b28f-90f8-4edf-b584-706f381c9d29" />
